sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -167,15 +240,50 @@ class ItemListElement extends Polymer.Element {
    * @param index index of item to select
    */
   _selectItem(index: number) {
-    this.set('rows.' + index + '.selected', true);
+    this._changeSelectItem(index, true);
   }
 
   /**
    * Unselects an item in the list.
    * @param index index of item to unselect
    */
   _unselectItem(index: number) {
-    this.set('rows.' + index + '.selected', false);
+    this._changeSelectItem(index, false);
+  }
+
+  /**
+   * Updates the show-details flag for a row after selection change.
+   */
+  _changeSelectItem(index: number, newValue: boolean) {
+    const previousSelectedCount =
```

Comment:
nit: I'd move these three lines down when they're needed in the `if` condition, and separate that section by the comment down there.